src/groups/group.jl

```diff
@@ -554,6 +554,215 @@ function inverse_translate_diff!(
     return translate_diff!(G, vout, inv(G, x), y, v, conv)
 end
 
+@doc doc"""
+    group_exp(G::AbstractGroupManifold, v)
+
+Compute the group exponential of the Lie algebra element `v`.
+
+Given an element $v ∈ 𝔤 = T_e G$, where $e$ is the [`identity`](@ref) element of the group
+$G$, and $𝔤$ is its Lie algebra, the group exponential is the map
+
+````math
+\exp_e : 𝔤 → G,
+````
+such that for $t ∈ ℝ$, $γ(t) = \exp_e (t v)$ defines a one-parameter subgroup with the
+following properties:
+
+````math
+\begin{aligned}
+γ(t) &= γ(-t)^{-1}\\
+γ(t + s) &= γ(t) \circ γ(s) = γ(s) \circ γ(t)\\
+γ(0) &= e\\
+\lim_{t → 0} \frac{d}{dt} γ(t) &= v.
+\end{aligned}
+````
+
+In general, the group exponential is distinct from the Riemannian exponential [`exp`](@ref).
+
+    group_exp(G::AbstractGroupManifold{AdditionOperation}, v)
+
+Compute $y = v$.
+
+    group_exp(G::AbstractGroupManifold{MultiplicationOperation}, v)
+
+For number and `AbstractMatrix` types of `v`, compute the usual numeric/matrix exponential,
+
+````math
+\exp_e = \sum_{n=0}^\infty \frac{1}{n!} v^n.
+````
+"""
+group_exp(M::Manifold, v) = group_exp(M, v, is_decorator_manifold(M))
+group_exp(M::Manifold, v, ::Val{true}) = group_exp(M.manifold, v)
```

Comment:
```suggestion
group_exp(M::Manifold, v, ::Val{true}) = group_exp(base_manifold(M), v)
```
We should probably use `base_manifold` in such places.

Reply:
The issue here is that `base_manifold` strips away all of the decorators, including `GroupManifold` and is therefore only appropriate for functions that are defined on the topological manifold. But I see now that it would probably be consistent to use the `is_decorator_group`/`base_group` pattern used for the other group-specific manifolds defined above.

Reply:
Right, `is_decorator_group`/`base_group` is the right solution :+1:

Reply:
After looking into this further I was wrong, the `is_decorator_group`/`base_group` pattern is reserved for interactions with the `Identity` element, which only makes sense in the context of its corresponding `AbstractGroupManifold`. The group-specific functions are designed so that they can be called on any manifold, though in general, they will only be implemented for `AbstractGroupManifold`. This lets one call e.g. `compose` or `group_exp` on a `ProductManifold`, which will of course error if any submanifolds are not groups but work just fine if they all are.

So for consistency, I'd like to keep this as written here. Do you have specific concerns with that? As far as I can tell, stripping off one decorator at a time this way until an implementation is reached has no more overhead than our `base_manifold` function does, and we already require that decorators have a `manifold` field that contains the decorated manifold.

Reply:
This was just a minor concern that maybe we shouldn't require all manifold decorators to have a `.manifold` field. We can keep the current version.

Reply:
I agree. The Julian way would be to have a function like `decorated_manifold(M::Manifold) = M.manifold` that can be overloaded if one used a different field name or maybe even doesn't store the decorated manifold within the decorator. We can handle that in a future PR though.